### DISCUSSION TITLE
Toggle main window minimization

### DISCUSSION THREAD

**Discussion by LinuxOnTheDesktop** [mullvad/mullvadvpn-app#9094] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

The main window appears when I click the tray icon (on Windows, at least; I have not tried the app on other OSs). But that window does not disappear - hide itself to the tray - when I click the tray icon for a second time. and I would like it to do so. For, I need to be able to hide the window quickly, for two reasons.

1. The tray icon at question resembles another icon in my tray, and thus _sometimes I show the Mullvad main window by accident_ and want very quickly to close that window;
2. _Sometimes I wish quickly to display, and then stop displaying, the information that the window shows_.

Also: 

3. the toggling behaviour that I request is _intuitive_.

Further:

4. I think that _your 'Better privacy/anonymity' check-box applies_.

### Alternative solutions

Re 2, above: one can use e.g. Rainmeter to display the information at issue, but (a) setting that up is non-trivial, and (b) sometimes one does want to check the app itself.

### Type of feature

- [X] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [X] Easier to use
- [ ] Other

### Operating System

- [ ] Android
- [ ] iOS
- [X] Windows
- [ ] macOS
- [ ] Linux

EDITED to fix typographical errors, and to improve punctuation.

**Comment by LinuxOnTheDesktop** [comment]
Why not fix this small paper-cut? It is an easy if small 'win'.

**Comment by FriederHannenheim** [comment]
I agree this would be pretty cool to have also on Linux

**Reply by FriederHannenheim** [reply]
Created a PR for this: https://github.com/mullvad/mullvadvpn-app/pull/9544

**Comment by LinuxOnTheDesktop** [comment]
I do not understand why this problem is not fixed. Someone has even been good enough to create a pull request (see above) - albeit one that seems buggy.

**Comment by Serock3** [comment]
Hi 🙂 I just wanted to chime in and let you know that, on Windows, left clicking the tray icon does toggle it, but only if the app window is not unpinned from the taskbar. It can be a little slow to react, though.

On Linux, left clicking the tray icon only opens the app. It might be a good idea to keep it consistent. We'll look into it!

**Comment by LinuxOnTheDesktop** [comment]
@Serock3 

Thanks for the comment. I find, on Windows, that _sometimes_ - perhaps only when I double click - does left-clicking the unpinned window toggle with the window.

**Comment by dlon** [comment]
This has been implemented, but it won't be released for a while. Likely in 2026.5.

**Comment by markus-mullvad** [comment]
Closing this as it has been implemented and will be included in a future release.
